addons/account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -108,128 +126,135 @@ def _check_parsed_data(self, cr, uid, stmts_vals, context=None):
+                raise UserError(_('The currency of the bank statement is not '
+                                'the same as the currency of the journal !'))
 
         # If we couldn't find/create a journal, everything is lost
         if not journal_id:
-            raise UserError(_('Cannot find in which journal import this statement. Please manually select a journal.'))
+            raise UserError(_('Cannot find in which journal import this '
+                            'statement. Please manually select a journal.'))
 
         return journal_id
 
-    def _create_journal(self, cr, uid, currency_id, account_number, context=None):
-        """ Create a journal and its account """
-        wmca_pool = self.pool.get('wizard.multi.charts.accounts')
-        company = self.pool.get('res.users').browse(cr, uid, uid, context=context).company_id
-
-        vals_account = {'currency_id': currency_id, 'acc_name': account_number, 'account_type': 'bank', 'currency_id': currency_id}
-        vals_account = wmca_pool._prepare_bank_account(cr, uid, company, vals_account, context=context)
-        account_id = self.pool.get('account.account').create(cr, uid, vals_account, context=context)
-
-        vals_journal = {'currency_id': currency_id, 'acc_name': _('Bank') + ' ' + account_number, 'account_type': 'bank'}
-        vals_journal = wmca_pool._prepare_bank_journal(cr, uid, company, vals_journal, account_id, context=context)
-        return self.pool.get('account.journal').create(cr, uid, vals_journal, context=context)
-
-    def _create_bank_account(self, cr, uid, account_number, journal_id=False, partner_id=False, context=None):
+    @api.model
+    @api.returns('res.partner.bank')
+    def _create_bank_account(self, account_number, journal_id=False):
         try:
-            type_model, type_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'base', 'bank_normal')
-            type_id = self.pool.get('res.partner.bank.type').browse(cr, uid, type_id, context=context)
-            bank_code = type_id.code
+            bank_type = self.env.ref('base.bank_normal')
+            bank_code = bank_type.code
         except ValueError:
             bank_code = 'bank'
-        account_number = account_number.replace(' ', '').replace('-', '')
         vals_acc = {
             'acc_number': account_number,
             'state': bank_code,
         }
-        # Odoo users bank accounts (which we import statement from) have company_id and journal_id set
-        # while 'counterpart' bank accounts (from which statement transactions originate) don't.
-        # Warning : if company_id is set, the method post_write of class bank will create a journal
+        # Odoo users bank accounts (which we import statement from) have
+        # company_id and journal_id set while 'counterpart' bank accounts
+        # (from which statement transactions originate) don't.
+        # UserError : if company_id is set, the method post_write of class
```

Comment:
when reading ```UserError``` in the docstring, one would expect to raise an UserError if some condition occures. It should be replaced by ```Note: if company_id is set...```